cli/cmd/deploy.go

```diff
@@ -1,47 +1,63 @@
 package cmd
 
 import (
+	"fmt"
 	"os"
 
 	"github.com/pchico83/i2kit/cli/providers/aws"
+	"github.com/pchico83/i2kit/cli/schemas/environment"
 	"github.com/pchico83/i2kit/cli/schemas/service"
 	"github.com/spf13/cobra"
 )
 
 //Deploy deploys an i2kit service
 func Deploy() *cobra.Command {
-	var file string
-	var space string
-	var dryRun bool
+	var servicePath string
+	var environmentPath string
 	cmd := &cobra.Command{
 		Use:   "deploy",
 		Short: "Deploy an i2kit service",
 		RunE: func(cmd *cobra.Command, args []string) error {
-			if err := validateEnvironment(); err != nil {
+			reader, err := os.Open(servicePath)
+			if err != nil {
 				return err
 			}
-			if err := initManifest(file); err != nil {
+			if err = service.Validate(reader); err != nil {
 				return err
 			}
-
-			reader, err := os.Open(file)
+			reader, err = os.Open(servicePath)
 			if err != nil {
 				return err
 			}
-
 			s, err := service.Read(reader)
 			if err != nil {
 				return err
 			}
-			config, err := getAWSConfig()
+
+			reader, err = os.Open(environmentPath)
+			if err != nil {
+				return err
+			}
+			if err = environment.Validate(reader); err != nil {
+				return err
+			}
+			reader, err = os.Open(environmentPath)
 			if err != nil {
 				return err
 			}
-			return aws.Deploy(s, space, config, dryRun)
+			e, err := environment.Read(reader)
+			if err != nil {
+				return err
+			}
+
+			if e.Provider == nil {
+				fmt.Println("Service dry-run deployed succesfully. Did you define an 'environment.yml' file?")
+				return nil
+			}
+			return aws.Deploy(s, e)
 		},
 	}
-	cmd.Flags().StringVarP(&file, "file", "f", "service.yml", "Service yml file to be deployed")
-	cmd.Flags().StringVarP(&space, "space", "s", "", "subdomains for dns search configuration")
-	cmd.Flags().BoolVarP(&dryRun, "dry-run", "", false, "True to fake i2kit deployments")
```

Comment:
@rberrelleza `dry-run` is not executed by passing no `Provider` section in `environment.yml`